### PR TITLE
Cleanup AWS service configs

### PR DESCRIPTION
**Pull request by reachjainrahul**
## Description
In AWS, we support only one service config. 

## Changes
Cleaning up the code to cater to that. Also, removed unnecessary service config common struct.

### CODE REVIEW

File: pkg/cloudprovider/cloudapi/aws/aws_security.go [hunk 1]
```diff
@@ -892,13 +872,8 @@ func (c *awsCloud) GetEnforcedSecurity() []securitygroup.SynchronizationContent
...
 
-			serviceCfg, err := accCfg.GetServiceConfigByName(awsComputeServiceNameEC2)
-			if err != nil {
-				awsPluginLogger().Error(err, "Enforced-security-cloud-view GET for account skipped", "account", accCfg.GetNamespacedName())
-				return
-			}
-			ec2Service := serviceCfg.(*ec2ServiceConfig)
-			err = ec2Service.waitForInventoryInit(inventoryInitWaitDuration)
+			ec2Service := accCfg.GetServiceConfig().(*ec2ServiceConfig)
+			err := ec2Service.waitForInventoryInit(inventoryInitWaitDuration)
```
Comment (Anandkumar26) [inline pkg/cloudprovider/cloudapi/aws/aws_security.go:876]: nit: Change this to if err (); err != nil {}

File: pkg/cloudprovider/cloudapi/azure/azure_security.go [hunk 2]
```diff
@@ -1075,13 +1058,8 @@ func (c *azureCloud) GetEnforcedSecurity() []securitygroup.SynchronizationConten
...
 
-			serviceCfg, err := accCfg.GetServiceConfigByName(azureComputeServiceNameCompute)
-			if err != nil {
-				azurePluginLogger().Error(err, "nforced-security-cloud-view GET for account skipped", "account", accCfg.GetNamespacedName())
-				return
-			}
-			computeService := serviceCfg.(*computeServiceConfig)
-			err = computeService.waitForInventoryInit(inventoryInitWaitDuration)
+			computeService := accCfg.GetServiceConfig().(*computeServiceConfig)
+			err := computeService.waitForInventoryInit(inventoryInitWaitDuration)
```
Comment (Anandkumar26) [inline pkg/cloudprovider/cloudapi/azure/azure_security.go:1062]: same as above.

File: pkg/cloudprovider/cloudapi/internal/accounts_common.go [hunk 3]
```diff
@@ -117,66 +109,37 @@ func (c *cloudCommon) updateCloudAccountConfig(client client.Client, credentials
...
-	for name, svcConfig := range existingSvcConfigMap {
-		newSvcCfg, found := newSvcConfigMap[name]
-		if !found {
-			// should not happen
-			continue
-		}
-		if err := svcConfig.updateServiceConfig(newSvcCfg); err != nil {
-			return err
-		}
-		logger.Info("Service config updated (api-clients to use new creds)", "account", accCfg.namespacedName,
```
Comment (Anandkumar26) [inline pkg/cloudprovider/cloudapi/internal/accounts_common.go:146]: Do we have  log message to know if a service config is updated or not?

File: pkg/cloudprovider/cloudapi/internal/accounts_common.go [hunk 4]
```diff
@@ -117,66 +109,37 @@ func (c *cloudCommon) updateCloudAccountConfig(client client.Client, credentials
...
 func (accCfg *cloudAccountConfig) performInventorySync() error {
 	accCfg.mutex.Lock()
 	defer accCfg.mutex.Unlock()
 
-	serviceConfigs := accCfg.serviceConfigs
-
 	ch := make(chan error)
 	var wg sync.WaitGroup
-	wg.Add(len(serviceConfigs))
+	wg.Add(1)
```
Comment (Anandkumar26) [inline pkg/cloudprovider/cloudapi/internal/accounts_common.go:125]: Do we still need a new thread for this?

File: pkg/cloudprovider/cloudapi/internal/accounts_common.go (hunk 3 above)
Comment (reachjainrahul) [reply pkg/cloudprovider/cloudapi/internal/accounts_common.go:146]: should be ok

File: pkg/cloudprovider/cloudapi/internal/accounts_common.go (hunk 4 above)
Comment (reachjainrahul) [reply pkg/cloudprovider/cloudapi/internal/accounts_common.go]: Done

**Review by Anandkumar26 — APPROVED** [review]
/LGTM

**Pull request merged**